python-package/lightgbm/arrow.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+"""Utilities for handling Arrow in LightGBM."""
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Iterator, Union
+
+import pyarrow as pa
+from pyarrow.cffi import ffi
+
+
+@dataclass
+class ArrowCArray:
```

Comment:
```suggestion
class _ArrowCArray:
```

Since this is intended to be an internal implementation detail, please prefix it with a `_`.